W-16264748-landing-page-general-use-fa (#250)
mulesoft/docs-anypoint-cli: modules/ROOT/pages/index.adoc

= Anypoint Platform CLI 4.x Overview
:page-aliases: runtime-manager::anypoint-platform-cli.adoc

Anypoint Platform CLI is a scripting and command-line tool for Anypoint Platform and Anypoint Platform Private Cloud Edition (Anypoint Platform PCE).

include::anypoint-cli::partial$intro.adoc[tag=usageintro]

With Anypoint CLI 4.x you can:
* Enhance your workflow efficiency: +
Streamline your API management and development processes with command-line tools.

> * ENsure consistency across environments and organizations: +

Reviewer: ```suggestion
* Ensure consistency across environments and organizations: +
```

Apply consistent configurations and policies across different organizations and environments, ensuring stability and reliability in your operations.
* Manage your applications effectively: +
Perform a wide range of operations on your APIs and applications, from deployment to monitoring, all from the command line.

WARNING: Anypoint Platform CLI 4.x is not backward compatible. If you are using a version of Anypoint CLI earlier than Anypoint CLI 4.x, xref:diff-earlier-ver.adoc[read this first to find out about the differences.]

== Before you begin 

> The Anypoint CLI is built with Node.js and is installed by using npm, the package manager for the Node JavaScript platform.
> You must install Anypoint CLI from npm before you can run the commands. 

Reviewer: ```suggestion
Because the Anypoint CLI is built with Node.js, you must:

. Install the Node package manager (npm)
. Install the Anypoint CLI using npm
+
After the Anypoint CLI installation, you can run the commands. 
```
This is just a suggestion, apply if you agree with it.

* If you are using Anypoint Platform PCE, pass the address where you are hosting the platform.
* If you are using Anypoint Platform EU Cloud, pass your EU control plane URL.

You can also pass this value using a dedicated environment variable `ANYPOINT_HOST=<name>`. For example, `ANYPOINT_HOST=my.anypoint.mulesoft.com`.

|===

[[overrides]]
== Override Order

You can use environment variables to define values in Anypoint CLI. Following is the override order if you mix environment variable settings with explicit command line values:

* Environment variables override credentials file parameters.
* Command-line parameters override environment variables. 
* If you do not pass a command-line flag, the default profile properties are used.
* If you do not specify an environment, the default is production.

[[session-timeout]]
== Session Timeout
Your Anypoint Platform session expires based on the *Default session timeout* configured in your Organization settings.

For information about Root Organization settings, see xref:access-management::organization.adoc#manage-root-organization-settings[Manage Root Organization Settings].

[[data-collection]]
== Anonymized Usage Data

> To improve the Anypoint CLI experience, we gather anonymized usage data. You can opt out by setting `collectMetrics` to false in the `credentials` file.

Reviewer: ```suggestion
To improve the Anypoint CLI experience, MuleSoft gathers anonymized usage data. You can opt out by setting `collectMetrics` to false in the `credentials` file.
```
